<launch>

<param command="$(find xacro)/xacro $(find merlin_description)/urdf/merlin.xacro" name="robot_description"/>

<!-- Load joint controller configurations from YAML file to parameter server -->
<rosparam file="$(find merlin_hardware_interface)/config/hardware.yaml" command="load"/>
<rosparam file="$(find merlin_hardware_interface)/config/controllers.yaml" command="load"/>
<rosparam file="$(find merlin_hardware_interface)/config/joint_limits.yaml" command="load"/>
  
<node name="merlin_hw" pkg="merlin_hardware_interface" type="merlin_hardware_interface_node" output="screen"/>

<node name="controller_spawner" output="screen" pkg="controller_manager" respawn="false" type="spawner"
  ns="/merlin" args="arm_controller joint_state_controller"/>

<!-- convert joint states to TF transforms for rviz, etc -->
<node name="robot_state_publisher" output="screen" pkg="robot_state_publisher" respawn="false" type="robot_state_publisher">
  <remap from="/joint_states" to="/merlin/joint_states"/>
</node>

</launch>